causalpy/pymc_meta_learners.py

```diff
@@ -0,0 +1,175 @@
+import pandas as pd
+import numpy as np
+import pymc as pm
+import xarray as xa
+
+from causalpy.utils import _fit
+from causalpy.skl_meta_learners import MetaLearner, SLearner, TLearner, XLearner, DRLearner
+from causalpy.pymc_models import LogisticRegression
+
+class BayesianMetaLearner(MetaLearner):
+    "Base class for PyMC based meta-learners."
+
+    def fit(self, X: pd.DataFrame, y: pd.Series, treated: pd.Series, coords=None):
+        "Fits model."
+        raise NotImplementedError()
+    
+
+
+
+class BayesianSLearner(SLearner, BayesianMetaLearner):
+    "PyMC version of S-Learner."
+
+    def predict_cate(self, X: pd.DataFrame) -> np.array:
+        X_untreated = X.assign(treatment=0)
+        X_treated = X.assign(treatment=1)
+        m = self.models["model"]
+
+        pred_treated = m.predict(X_treated)["posterior_predictive"].mu
+        pred_untreated = m.predict(X_untreated)["posterior_predictive"].mu
+
+        cate = pred_treated - pred_untreated
+
+        return cate
+
+
+class BayesianTLearner(TLearner, BayesianMetaLearner):
+    "PyMC version of T-Learner."
+
+    def predict_cate(self, X: pd.DataFrame) -> np.array:
+        treated_model = self.models["treated"]
+        untreated_model = self.models["untreated"]
+
+        pred_treated = treated_model.predict(X)["posterior_predictive"].mu
+        pred_untreated = untreated_model.predict(X)["posterior_predictive"].mu
+
+        cate = pred_treated - pred_untreated
+
+        return cate
+
+
+class BayesianXLearner(XLearner, BayesianMetaLearner):
+    "PyMC version of X-Learner."
+
+    def __init__(
+        self,
+        X,
+        y,
+        treated,
+        model=None,
+        treated_model=None,
+        untreated_model=None,
+        treated_cate_estimator=None,
+        untreated_cate_estimator=None,
+        propensity_score_model=LogisticRegression()
```

Comment:
Do you want to allow the `propensity_score_model` to be a bayesian model as well?